dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java

```diff
@@ -82,12 +79,80 @@ public void shouldSaveCondition() throws DotSecurityException, DotDataException
         final Rule rule = new RuleDataGen().host(host).nextPersisted();
 
         final ConditionGroup conditionGroup = new ConditionGroupDataGen().rule(rule).next();
-        this.addPermission(role, host, true);
+        this.addRulesPermission(role, host, true);
 
         //Saving and testing GroupCondition
         rulesAPI.saveConditionGroup(conditionGroup, user, false);
 
         List<Rule> allRules = rulesAPI.getAllRules(user, false);
+        checkCondition(user, conditionGroup, allRules);
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#saveConditionGroup(ConditionGroup, User, boolean)}
+     *                  and {@link RulesAPIImpl#saveCondition(Condition, User, boolean)}
+     * When:Try to create a Group Condition and a Condition in a page's rule with PuBLISH permission over the page
+     * Should: save the rule
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSaveConditionInRulesPage() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+
+        final Template template = new TemplateDataGen().nextPersisted();
+        final HTMLPageAsset htmlPageAsset = new HTMLPageDataGen(host, template)
+                .nextPersisted();
+
+        addPermission(role, htmlPageAsset, PermissionAPI.PERMISSION_PUBLISH);
+        addRulesPermission(role, host, true);
+        final Rule rule = new RuleDataGen().page(htmlPageAsset).nextPersisted();
+
+        final ConditionGroup conditionGroup = new ConditionGroupDataGen().rule(rule).next();
+        this.addRulesPermission(role, host, true);
+
+        //Saving and testing GroupCondition
+        rulesAPI.saveConditionGroup(conditionGroup, user, false);
+
+        List<Rule> allRules = rulesAPI.getAllRules(user, false);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'allRules' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)